File: backend/src/main/java/org/akashihi/mdg/entity/Rate.java
package org.akashihi.mdg.entity;


import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.Table;
import java.math.BigDecimal;
import java.time.LocalDateTime;

@Getter
@Setter
@ToString
@Entity
@Table(name = "rates")
@NoArgsConstructor
@AllArgsConstructor
public class Rate {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @Column(name = "rate_beginning")
    private LocalDateTime beginning;
    @Column(name = "rate_end")
    private LocalDateTime end;
    @Column(name = "from_id")
    private Long from;
    @Column(name = "to_id")
    private Long to;
    private BigDecimal rate;
}

File: backend/src/main/kotlin/org/akashihi/mdg/entity/Rate.kt
package org.akashihi.mdg.entity

import com.fasterxml.jackson.databind.annotation.JsonSerialize
import org.akashihi.mdg.api.v1.json.TsSerializer
import java.math.BigDecimal
import java.time.LocalDateTime
import javax.persistence.Column
import javax.persistence.Entity
import javax.persistence.GeneratedValue
import javax.persistence.GenerationType
import javax.persistence.Id
import javax.persistence.Table

@Entity
@Table(name = "rates")
class Rate(
    @Column(name = "rate_beginning")
    @field:JsonSerialize(using = TsSerializer::class)
    var beginning: LocalDateTime,

    @Column(name = "rate_end")
    @field:JsonSerialize(using = TsSerializer::class)
    var end: LocalDateTime,

    @Column(name = "from_id")
    val from: Long,

    @Column(name = "to_id")
    val to: Long,
    val rate: BigDecimal,

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = null
)
